src/Hooks/CheckoutScriptHooks.php

```diff
@@ -199,7 +199,12 @@ private function shouldShowDeliveryOptions(): bool
             /**  @var WC_Product $product */
             $product = $cartItem['data'];
 
-            if (! $product->is_virtual() && ! $product->is_on_backorder($cartItem['quantity'])) {
+            if (! $product->is_virtual()
+                && (! $product->is_on_backorder($cartItem['quantity'])
+                    || Settings::get(
```

Comment:
Hoewel dit wel goed is denk ik, zou ik de loop toch wel optimaliseren hier, door `Settings::get` in een var te stoppen voor de loop, zodat het niet bij elk product opgehaald hoeft te worden, en dan te checken als eerste conditie van de `||` statement, evt. besparen op aanroepen van `->is_on_backorder`, of misschien zie jij het anders?

Reply:
Daar ben ik het mee eens. Ik heb de nodige aanpassingen doorgevoerd.